src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -382,18 +382,38 @@ private ConfigurationManager<OpenIdConnectConfiguration> CreateConfigManager(
             }
         }
 
-        private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, string actualIssuer)
+        internal static bool IsValidIssuer(string issuerTemplate, string tenantId, string tokenIssuer)
         {
-            if (string.IsNullOrEmpty(validIssuerTemplate))
+            if (string.IsNullOrEmpty(issuerTemplate) || string.IsNullOrEmpty(tenantId) || string.IsNullOrEmpty(tokenIssuer))
                 return false;
 
-            if (validIssuerTemplate.Contains(TenantIdTemplate))
+            ReadOnlySpan<char> issuerTemplateSpan = issuerTemplate.AsSpan();
+            ReadOnlySpan<char> tokenIssuerSpan = tokenIssuer.AsSpan();
+            int templateTenantIdPosition = issuerTemplate.IndexOf(TenantIdTemplate, StringComparison.Ordinal);
+
+            // If the template contains the tenantIdTemplate, ensure the actual issuer matches the template with the tenantId replaced.
+            if (templateTenantIdPosition >= 0 && tokenIssuer.Length > templateTenantIdPosition)
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                // Ensure the prefix of the issuer template matches the token issuer's prefix
+                if (!issuerTemplateSpan.Slice(0, templateTenantIdPosition).SequenceEqual(tokenIssuerSpan.Slice(0, templateTenantIdPosition)))
+                    return false;
+
+                // Ensure tokenIssuer is atleast as long as issuerTemplate with tenantIdTemplate replaced
+                if (tokenIssuer.Length <= templateTenantIdPosition + tenantId.Length)
```

Comment:
Is there a test case for this `if` block?

Reply:
`Malformed_V2_TokenIssuer_Failure` is the test for the if block where the tokenIssuer is malformed and its length is less than the issuerTemplate with tenantId replaced.